src/zmq/zmqnotificationinterface.cpp

```diff
@@ -29,6 +33,15 @@ CZMQNotificationInterface::~CZMQNotificationInterface()
     }
 }
 
+void CZMQNotificationInterface::AddJSONNotifiers(UniValue& publish) const
+{
+    for (const auto* n : notifiers) {
+        const std::string& type = n->GetType();
+        assert(type.substr(0, 3) == "pub");
```

Comment:
Why assert?

Reply:
Because the current code is meant to only add types that start with `"pub"` - if that is not the case, it is a bug with the code (or at least means the code here has to be updated as well).

But if you prefer, we could just ignore those that don't start with `"pub"` and print a warning to `debug.log` instead.   Or we could just keep the `"pub"` prefix in the output - whatever you think is best.